apps/cli/src/commands/build.ts

```diff
@@ -145,111 +145,94 @@ Update your application Dockerfile using one of the templates at https://github.
         return info;
     }
 
-    private async startBuildContainer(
-        appImage: string,
-        builderImage: string,
-    ): Promise<string> {
+    // creates a rootfs tarball from the image
+    // this process is not always fully reproducible
+    // FIXME: we could use the image and create a flat rootfs without
+    //        `docker container create` (use undocker, umoci, a native typescript implementation, etc.)
+    private async createTarball(
+        image: string,
+        outputFilePath: string,
+    ): Promise<void> {
         // create docker tarball from app image
         const { stdout: appCid } = await execa("docker", [
             "container",
             "create",
             "--platform",
             "linux/riscv64",
-            appImage,
+            image,
         ]);
 
         await execa("docker", [
             "container",
             "export",
             "-o",
-            SUNODO_DEFAULT_TAR_PATH,
+            outputFilePath,
             appCid,
         ]);
 
-        // start build container
-        const { stdout: cid } = await execa("docker", [
+        await execa("docker", ["container", "rm", appCid]);
+    }
+
+    // this wraps the call to the sdk image with a one-shot approach
+    // the (inputPath, outputPath) signature will mount the input as a volume and copy the output with docker cp
+    private async sdkRun(
+        sdkImage: string,
+        cmd: string[],
+        inputPath: string,
+        outputPath: string,
+    ): Promise<void> {
+        const randomPrefix = Math.random().toString(36).substring(7);
+        const containerName = `sdk-run-${randomPrefix}`;
```

Comment:
You don’t need a random container name. Get the container id as we do above

Reply:
but we get the container id from the `docker container create` command

To get the container ID from `docker container run` I'd have to use `--cidfile=` and read from the file, seems more work than just defining a random value at runtime. What do you think?

Reply:
sent a fixup